packages/contracts-core/contracts/Replica.sol

```diff
@@ -186,6 +186,7 @@ contract Replica is Version0, NomadBase {
         bytes32 _messageHash = _m.keccak();
         require(acceptableRoot(messages[_messageHash]), "!proven");
         // check re-entrancy guard
+        entered = 1;
```

Comment:
this breaks the re-entrancy guard entirely

Reply:
Without that, my tests completely broke at that point. I remember thinking that it makes sense. Let me look at it again.

Reply:
Wait, that's indeed retarded. wtf